Xamarin.Forms.Core/Shell/MenuShellItem.cs

```diff
@@ -9,8 +9,8 @@ internal MenuShellItem(MenuItem menuItem)
 			MenuItem = menuItem;
 
 			SetBinding(TitleProperty, new Binding("Text", BindingMode.OneWay, source: menuItem));
-			SetBinding(IconProperty, new Binding("Icon", BindingMode.OneWay, source: menuItem));
-			SetBinding(FlyoutIconProperty, new Binding("Icon", BindingMode.OneWay, source: menuItem));
+			SetBinding(IconProperty, new Binding("IconImageSource", BindingMode.OneWay, source: menuItem));
```

Comment:
If I can make a further suggestion, maybe these bindings should use `MenuItem.IconImageSourceProperty.PropertyName` instead of just magic strings. If that had been done in the first place this issue never would have happened.